src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -109,17 +112,27 @@ export default class ConfirmOrderPlugin extends Plugin {
 
     renderPaymentComponent(type) {
         if (type === 'oneclick') {
-            this.renderStoredPaymentMethodComponents()
+            this.renderStoredPaymentMethodComponents();
             return;
         }
+
+        let identifier = 'type';
+        // Filter payment method configs by brand in the case of giftcards
+        if (adyenCheckoutOptions.selectedPaymentMethodHandler.includes('giftcard')) {
+            identifier = 'brand';
+        }
+
         // Get the payment method object from paymentMethodsResponse
-        let filtered = $.grep(this.adyenCheckout.paymentMethodsResponse.paymentMethods, function(paymentMethod) {
-            return paymentMethod.type === type;
+        let paymentMethodConfigs = $.grep(this.adyenCheckout.paymentMethodsResponse.paymentMethods, function(paymentMethod) {
+            return paymentMethod[identifier] === type;
         });
-        if (filtered.length === 0) {
+        if (paymentMethodConfigs.length === 0) {
+            if (this.adyenCheckout.options.environment === 'test') {
+                console.error('Payment method configuration not found. ', state);
```

Comment:
Should we add a visible error message here as well?

Reply:
We have not yet implemented error messages in the frontend, but yes this is a good idea